heron/packing/src/java/com/twitter/heron/packing/roundrobin/RoundRobinPacking.java

```diff
@@ -343,4 +351,37 @@ private void validatePackingPlan(PackingPlan plan) throws PackingException {
       }
     }
   }
+
+  /*
+   * read the current packing plan with update parallelism to calculate a new packing plan
+   * the packing algorithm packInternal() is shared with pack()
+   * delegate to packInternal() with the new container count and component parallelism
+   */
+  @Override
+  public PackingPlan repack(PackingPlan currentPackingPlan, Map<String, Integer> componentChanges)
+      throws PackingException {
+    int initialNumContainer = TopologyUtils.getNumContainers(topology);
+    int initialNumInstance = TopologyUtils.getTotalInstance(topology);
+    double initialNumInstancePerContainer = (double) initialNumInstance / initialNumContainer;
+
+    Map<String, Integer> currentComponentParallelism = currentPackingPlan.getComponentCounts();
+    Map<String, Integer> newComponentParallelism = new HashMap<>();
```

Comment:
I am wondering if it is easier to understand this way:

newComponentParallelism = currentComponentParallelism;
for (e: componentChanges) {
  newComponentParallelism.put(e.getKey, e.getValue);
}

int newNumInstance = TopologyUtils.getTotalInstance(newComponentParallelism);

Reply:
also, I am wondering if we should verify that the value needs to be positive?

Reply:
newComponentParallelism is supposed to include the old and new components, while the suggested code includes new components only.

Reply:
added positive parallelism check in RuntimeManagerRunner.java

Reply:
adopt @nwangtw suggestion code